mne/transforms.py

```diff
@@ -399,25 +470,21 @@ def transform_surface_to(surf, dest, trans):
         Transformed source space. Data are modified in-place.
     """
     if isinstance(dest, string_types):
-        if dest not in _frame_dict:
+        if dest not in _str_to_frame:
             raise KeyError('dest must be one of %s, not "%s"'
-                           % (list(_frame_dict.keys()), dest))
-        dest = _frame_dict[dest]  # convert to integer
+                           % (list(_str_to_frame.keys()), dest))
+        dest = _str_to_frame[dest]  # convert to integer
     if surf['coord_frame'] == dest:
         return surf
 
-    if trans['to'] == surf['coord_frame'] and trans['from'] == dest:
-        trans = invert_transform(trans)
-    elif trans['from'] != surf['coord_frame'] or trans['to'] != dest:
-        raise ValueError('Cannot transform the source space using this '
-                         'coordinate transformation')
-
+    trans = _ensure_trans(trans, int(surf['coord_frame']), dest)
     surf['coord_frame'] = dest
-    surf['rr'] = apply_trans(trans['trans'], surf['rr'])
-    surf['nn'] = apply_trans(trans['trans'], surf['nn'], move=False)
+    surf['rr'] = apply_trans(trans, surf['rr'])
+    surf['nn'] = apply_trans(trans, surf['nn'], move=False)
     return surf
 
 
+@deprecated('transform_coordinates is deprecated and will be removed in v0.11')
```

Comment:
Turns out that `transform_coordinates` uses the `'meg'` coordinate frame to mean *head* coordinates, whereas everywhere else in our codebase it means *meg device* coordinates. Instead of fixing this, I think it's better just to remove it. I don't think it's really used, and there should be better / more explicit ways of accomplishing the same thing.